server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -881,6 +895,18 @@ public String toString() {
         });
     }
 
+    public Releasable withDiscoveryListener(Consumer<Iterable<DiscoveryNode>> listener) {
```

Comment:
These changes to `Coordinator` are tested by `testDiscoveryOfPeersTriggersNotification`.